File: 프로그래머스/2/60057. 문자열 압축/문자열 압축.java
class Solution {
    public int solution(String s) {
        int answer = 0;

        for(int i=1; i<=(s.length()/2)+1; i++){
            int result = getSplitedLength(s, i, 1).length();
            answer = i==1 ? result : (answer>result?result:answer);
        }

        return answer;
    }

    public String getSplitedLength(String s, int n, int repeat){
        if(s.length() < n) return s;
        String result = "";
        String preString = s.substring(0, n);
        String postString = s.substring(n, s.length());

        // 불일치 -> 현재까지 [반복횟수 + 반복문자] 조합
        if(!postString.startsWith(preString)){
            if(repeat ==1) return result += preString + getSplitedLength(postString, n, 1);
            return result += Integer.toString(repeat) + preString + getSplitedLength(postString, n, 1);
        }

        return result += getSplitedLength(postString, n, repeat+1);
    }
}

File: 프로그래머스/2/60057. 문자열 압축/문자열 압축.js
function solution(s) {
    var answer = Infinity
    
    // 문자열 길이 설정
    for(let i=1; i<=s.length; i++) {
        let res = ""
        let strPre = ""
        let cnt = 1
        
        for(let j=0; j<s.length; j+=i) {
            const strCur = s.substring(j,j+i)
            if(strCur === strPre) {
                cnt+=1
            } else {
                res += (cnt > 1 ? cnt.toString() : "") + strPre
                strPre = strCur
                cnt=1
            }
        }
        res += (cnt > 1 ? cnt.toString() : "") + strPre
        answer = Math.min(answer,res.length)
    }
    
    return answer;
}